
uint32 x_id
uint32 y_id
float32 loc_x
float32 loc_y